### PR TITLE
RestoreItemAction v1 refactoring for plugin api versioning

### PR DESCRIPTION
**Pull request by sseago**
Thank you for contributing to Velero!

This PR refactors the restoreitemaction code in preparation for plugin api versioning. With these changes, we will be ready to add the RIA v2 api definition and update restore to use v2 RIA plugins.

https://github.com/vmware-tanzu/velero/pull/5271 needs to be merged first before this one.

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by shubham-pampattiwar — DISMISSED** [review]

**Review by shubham-pampattiwar — DISMISSED** [review]

**Comment by sseago** [comment]
Hmm. The CI failure doesn't seem to be related to my PR (although I haven't dug into it deeply yet, so there may be an issue there -- but this passed fine before I rebased the PR to main this morning)
```
--- FAIL: TestGetVolumesRepositoryType (0.00s)
    --- FAIL: TestGetVolumesRepositoryType/mismatch_repository_type (0.00s)
        restorer_test.go:87: 
            	Error Trace:	restorer_test.go:87
            	Error:      	Error message not equal:
            	            	expected: "multiple repository type in one backup, current type fake-type1, differential one [type fake-type2, snapshot ID fake-snapshot-id-2, uploader fake-uploader-2]"
            	            	actual  : "multiple repository type in one backup, current type fake-type2, differential one [type fake-type1, snapshot ID , uploader ]"
            	Test:       	TestGetVolumesRepositoryType/mismatch_repository_type
```
Also, I just confirmed that this test is running fine in my local checkout of the PR branch. I'll re-run it just in case there's a nondeterminstic error in the tests.

@kaovilai Didn't you report on slack that you saw this error as well?

**Comment by sseago** [comment]
Also, I noticed that the refactored PR #5318 passed CI tests, even though that has all of the same code from this PR plus one new commit.

**Comment by kaovilai** [comment]
A few force pushes with some changes later and no longer hitting that error

**Comment by sseago** [comment]
@kaovilai Yeah -- there must be some sort of race condition or other nondeterministic failure case in that unit test. I'll create a github issue for it.

**Review by shubham-pampattiwar — DISMISSED** [review]

**Comment by sseago** [comment]
This passed on the third attempt to run the CI test (with no code changes). I've created https://github.com/vmware-tanzu/velero/issues/5352 to track the sporadic test failure issue.

File: pkg/plugin/clientmgmt/restoreitemaction/v1/shared_test.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
+/*
```
Comment (blackpiglet) [inline pkg/plugin/clientmgmt/restoreitemaction/v1/shared_test.go:1]: shared_test.go file is same with backupitemaction's.
Can we let all plugins use the same shared_test.go file?

Comment (sseago) [reply pkg/plugin/clientmgmt/restoreitemaction/v1/shared_test.go:1]: @blackpiglet unfortunately, the way the test framework works, we can't, at least the way it's currently implemented. _test files aren't exported in the package. We could create a util package somewhere and use that, although then could end up with test framework-related code in the main velero image. That may be OK, though. Basically if we want this to remain part of the tests only, then copying it into the varius packages will be needed. If we're ok making it a shared, exportable package, then we could refactor useage here (and in the other places it's used).

Comment (sseago) [reply pkg/plugin/clientmgmt/restoreitemaction/v1/shared_test.go]: @blackpiglet ok, I went ahead and made the changes. This code is now in internal/restartabletest and I've removed all copies of it (the one in this PR and the two already merged) and updated references to use the exported package instead.

**Review by blackpiglet — APPROVED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]

**Pull request merged**